hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1725,6 +1739,189 @@ public void delete(final Path path, final boolean recursive,
     } while (shouldContinue);
   }
 
+  /**
+   * Handles deletion of a path over Blob Endpoint.
+   * @param path path to be deleted
+   * @param recursive defines deletion to be recursive or not
+   * @param tracingContext object to trace the API flows
+   *
+   * @throws IOException exception from server call or if recursive used on
+   * non-empty directory
+   */
+  private void deleteBlobPath(final Path path,
+      final boolean recursive,
+      final TracingContext tracingContext) throws IOException {
+    StringBuilder listSrcBuilder = new StringBuilder();
+    String srcPathStr = path.toUri().getPath();
+    listSrcBuilder.append(srcPathStr);
+    if (!path.isRoot()) {
+      listSrcBuilder.append(FORWARD_SLASH);
+    }
+
+    String listSrc = listSrcBuilder.toString();
+    BlobList blobList = client.getListBlobs(null, listSrc, null, null,
+        tracingContext).getResult().getBlobList();
+    if (blobList.getBlobPropertyList().size() > 0) {
+      orchestrateBlobDirDeletion(path, recursive, listSrc, blobList, tracingContext);
+    } else {
+      LOG.debug(String.format("Path %s doesn't have child-blobs", srcPathStr));
+      if (!path.isRoot()) {
+        try {
+          LOG.debug(String.format("Deleting path %s", srcPathStr));
+          client.deleteBlobPath(path, null, tracingContext);
+          LOG.debug(String.format("Path deleted %s", srcPathStr));
```

Comment:
Logging is too verbose. Can this be cut down.

Reply:
+1

Reply:
I feel we should keep it. Reason being at this point we are on non-hierarchical path. Its only one delete call we are tracking via logs. Here chances of 404 raising are also there. Keeping it in logs I feel would be helpful while debugging via logs. Let me please know if you disagree with my reasoning. Thanks.